### ISSUE TITLE
GNOME crashes on login (Debian testing)

### ISSUE THREAD

**Issue by ethernetlord** [spin83/multi-monitors-add-on#144]
I'm using Debian testing/bullseye (GNOME 3.38.2, Multi Monitors addon 22-1 [from the Debian repo]).

When I login to GNOME shell with the extension enabled, the screen blinks a few times and then displays the "Oh no! Something has gone wrong!" screen. On the next login, all extensions get automatically disabled and I'm able to log in without any problems. If I enable the extension after logging in, it works perfectly fine.

This is what `journalctl` says:
```
pro 30 11:14:50 laptop gnome-shell[11991]: Enable Multi Monitors Add-On (v22)...
pro 30 11:14:50 laptop gnome-shell[11991]: Some code accessed the property 'DoNotDisturbSwitch' on the module 'calendar'. That property was defined with 'let' or 'const' inside the module. This was previously supported, but is not correct according to the ES6 standard. Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.
pro 30 11:14:50 laptop gnome-shell[11991]: new: i1x0y0w1920h1080
pro 30 11:14:50 laptop gnome-shell[11991]: pi:0
pro 30 11:14:50 laptop gnome-shell[11991]: i:0 x:1920 y:0 w:1920 h:1080
pro 30 11:14:50 laptop gnome-shell[11991]: i:1 x:0 y:0 w:1920 h:1080
pro 30 11:14:50 laptop dbus-daemon[11524]: [session uid=1000 pid=11524] Activating service name='org.gnome.Shell.Notifications' requested by ':1.71' (uid=1000 pid=11991 comm="/usr/bin/gnome-shell ")
pro 30 11:14:50 laptop systemd[2100]: Started GNOME Shell on X11.
pro 30 11:14:50 laptop dbus-daemon[11524]: [session uid=1000 pid=11524] Successfully activated service 'org.gnome.Shell.Notifications'
pro 30 11:14:50 laptop gnome-shell[11991]: Error looking up permission: GDBus.Error:org.freedesktop.portal.Error.NotFound: No entry for geolocation
pro 30 11:14:51 laptop gsd-media-keys[11827]: Failed to grab accelerator for keybinding settings:playback-random
pro 30 11:14:51 laptop gsd-media-keys[11827]: Failed to grab accelerator for keybinding settings:hibernate
pro 30 11:14:51 laptop gsd-media-keys[11827]: Failed to grab accelerator for keybinding settings:playback-repeat
pro 30 11:14:51 laptop gsd-media-keys[11827]: Failed to grab accelerator for keybinding settings:rotate-video-lock
pro 30 11:14:51 laptop gsd-media-keys[11827]: Failed to grab accelerator for keybinding settings:rfkill
pro 30 11:14:51 laptop gnome-shell[11991]: The program 'gnome-shell' received an X Window System error.
                                              This probably reflects a bug in the program.
                                              The error was 'BadValue (integer parameter out of range for operation)'.
                                                (Details: serial 574 error_code 2 request_code 138 (XFIXES) minor_code 31)
                                                (Note to programmers: normally, X errors are reported asynchronously;
                                                 that is, you will receive the error a while after causing it.
                                                 To debug your program, run it with the GDK_SYNCHRONIZE environment
                                                 variable to change this behavior. You can then get a meaningful
                                                 backtrace from your debugger if you break on the gdk_x_error() function.)
pro 30 11:14:51 laptop gnome-shell[11991]: == Stack trace for context 0x556a363732e0 ==
pro 30 11:14:51 laptop systemd[2100]: org.gnome.Shell@x11.service: Main process exited, code=killed, status=5/TRAP
pro 30 11:14:51 laptop systemd[2100]: org.gnome.Shell@x11.service: Failed with result 'signal'.
pro 30 11:14:51 laptop systemd[2100]: org.gnome.Shell@x11.service: Scheduled restart job, restart counter is at 2.
pro 30 11:14:51 laptop systemd[2100]: Stopped GNOME Shell on X11.
```

**Comment by Zappo-II** [comment]
Same with Manjaro - gnome 3.38.2 - kernel 5.4 / 5.9 / 5.10 and NVIDIA 455.45.01-6 using latest version from gnome.org in .local/share/gnome-shell/extensions...

**Comment by ethernetlord** [comment]
To be more exact, my uname is `Linux laptop 5.9.0-5-amd64 #1 SMP Debian 5.9.15-1 (2020-12-17) x86_64 GNU/Linux` and my X.org version is 1.20.8 (I'm not using Wayland). I face this issue on an AMD Ryzen "Renoir" integrated GPU, so it isn't a NVIDIA-only problem.

**Comment by internauta2000** [comment]
> Same with Manjaro - gnome 3.38.2 - kernel 5.4 / 5.9 / 5.10 and NVIDIA 455.45.01-6 using latest version from gnome.org in .local/share/gnome-shell/extensions...
> [edit] Using xorg-server 1.20.10

+ 1 exactly config..!

**Comment by AnthonySLWhite** [comment]
Same in Pop_OS 20.10
- GNOME: 3.38.1
- CPU: Intel® Core™ i7-5820K
- GPU: NVIDIA Corporation GM204 [GeForce GTX 970]

**Comment by StianOby** [comment]
I started experiencing the same today, on Ubuntu 20.04.

My uname is: Linux oby-desktop 5.8.0-40-generic #45-Ubuntu SMP Fri Jan 15 11:05:36 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Comment by DruidicBearClan** [comment]
Same for me, uname: 

5.8.0-40-generic #45-Ubuntu SMP Fri Jan 15 11:05:36 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Comment by netronicus** [comment]
Same error here on Manjaro 20.2.1

**Comment by spin83** [comment]
Looks like a duplicate of #140 issue.

**Comment by okennedy** [comment]
PopOS 20.10 here.  Running the current repository head resolved the issue.  As of mid-Jan, this version had not yet made it onto the gnome extensions page.

**Comment by tristobal** [comment]
Same error on Manjaro Nibia (v20.2.1).

**Comment by Zappo-II** [comment]
Got fixed for me on Manjaro (up to date Kernel 5.10 / nvidia-latest) by this commit ( https://github.com/spin83/multi-monitors-add-on/commit/f8343948ab835081f3be6987730decf592418e27 ) as mentioned in #140 ...
Thanxalot ant SRY for me beeing unaware that this one was already fixed at time complaining about it...

**Comment by ronki2304** [comment]
I have the same problem,

I found this log when I tried to open the parameter 

> The settings of extension multi-monitors-add-on@spin83 had an error:
```
Error: GSettings key show-thumbnails-slider not found in schema org.gnome.shell.extensions.multi-monitors-add-on
```

Stack trace:
```
_checkKey@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:580:23
createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:532:30
_addSettingsBooleanSwitch@/home/jeremy/.local/share/gnome-shell/extensions/multi-monitors-add-on@spin83/prefs.js:242:18
_addBooleanSwitch@/home/jeremy/.local/share/gnome-shell/extensions/multi-monitors-add-on@spin83/prefs.js:231:14
_init@/home/jeremy/.local/share/gnome-shell/extensions/multi-monitors-add-on@spin83/prefs.js:66:14
buildPrefsWidget@/home/jeremy/.local/share/gnome-shell/extensions/multi-monitors-add-on@spin83/prefs.js:251:18
_init@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:207:40
OpenExtensionPrefsAsync/<@resource:///org/gnome/Shell/Extensions/js/extensionsService.js:122:28
asyncCallback@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:132:22
run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:177:20
main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
run@resource:///org/gnome/gjs/modules/script/package.js:222:19
start@resource:///org/gnome/gjs/modules/script/package.js:206:8
@/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17
```

**Comment by doronmak** [comment]
same here using ubuntu 20.10

**Comment by Zappo-II** [comment]
Did anybody of you people who continue to report problems mind to take a look at #140 and read the README.MD of this repo especially the section about installing from git to ensure to use the current master of this repo...???

This issue should be fixed if you mind using this version...

```
tuxedo Linux 5.10.7-3-MANJARO x86_64 20.2.1 Nibia
~ >>> cat .local/share/gnome-shell/extensions/multi-monitors-add-on@spin83/metadata.json 
{
    "shell-version": ["3.38"],
    "uuid": "multi-monitors-add-on@spin83",
    "name": "Multi Monitors Add-On",
    "settings-schema": "org.gnome.shell.extensions.multi-monitors-add-on",
    "gettext-domain": "multi-monitors-add-on",
    "description": "Add multiple monitors overview and panel for gnome-shell.",
    "url": "https://github.com/spin83/multi-monitors-add-on.git",
    "version": 22.1
}
~ >>> 
```

**Comment by StianOby** [comment]
Fair point, unless one sees this issue as being not just a duplicate of #140, but rather a call for pushing the latest version to the GNOME Extensions website. But, sure, I or anyone else can always open a separate issue for that, and have this closed as duplicate of #140 (which I didn't spot at the time when I submitted my report above).
